[ES|QL] Improves the authoring of the query and charts

stratoula opened the pull request:
## Summary

It makes the authoring of the ES|QL query easier by:

- changing the flyout to be resizable
- displaying a tab with the results of the query in the ES|QL datagrid component.

<img width="1255" alt="image" src="https://github.com/user-attachments/assets/526509fa-7313-4560-9186-61181b5c575b">


### Checklist

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

andreadelrio commented:
Hi @stratoula, a very nice addition to our ES|QL Dashboard UX here! As requested by @teresaalvarezsoler here's a first round of design feedback I wanted to share. 

a) I don’t think we should be offering the entire power of Discover in such a small area. Therefore, I suggest we provide a simplified version of Discover here. 
- [ ] Remove the checkboxes and with them the `Selected` and `Compare` buttons and related functionalities
<img width="1440" alt="Frame 1" src="https://github.com/elastic/kibana/assets/4016496/60a552c5-4a16-4889-8d9c-7cf8665fc774">


- [ ] Offer a link for users to take their query and see it in the full Discover app if they wish to do so. Label it `Open in Discover`. See:
<img width="1440" alt="Frame 3" src="https://github.com/elastic/kibana/assets/4016496/ac9c8b09-9b11-4d11-8d56-0ec8967b8f7c">

- [ ] Given the limited space, I suggest we set `Single` as the default setting for **Cell row height**.
- [ ] For the `Pagination`, given the limited space where we are displaying the results I think we should allow 25 rows per pages at the most. So we’d have `Rows per page = [10, 25]`.

b) If I make any changes to the table, including resizing the columns, and then I edit and re-run the query, my changes to the table should not be altered. This is not the behavior currently. See:

![format_settings_list](https://github.com/elastic/kibana/assets/4016496/3b9655fe-9f27-4f0d-ab8f-bfd51589f4d7)

- [ ] Keep table settings (columns width, `Cell row height`, etc) intact upon editing and re-running the query

c) Other suggestions and/or findings

- [ ] Just like in Discover, make the table’s header sticky
- [ ] Similar to what we do for the `Suggestions` accordion, add a counter to the `Results` accordion
- [ ] Consistently show the `Columns` button. Right now, sometimes it disappears. See:
![columns_dissapear](https://github.com/elastic/kibana/assets/4016496/d75a7721-2bc2-447e-a7cd-40fc679dd529)

stratoula commented:
@andreadelrio thanx, not sure when I will find the time to work with this PR again but some comments: 

>  If I make any changes to the table, including resizing the columns, and then I edit and re-run the query, my changes to the table should not be altered.

This is true but as the same happens for viz changes I suggest to tackle them together. (I have a task for this in 8.17)

>  Consistently show the Columns button. Right now, sometimes it disappears. See:

My guess is that this is how Discover works in small spaces but I will take a look

jughosta commented:
> Consistently show the Columns button. Right now, sometimes it disappears.

That's a part of `EuiDataGrid` logic for smaller screens, but we could override it in `UnifiedDataTable` if necessary https://github.com/elastic/kibana/blob/339379e9e884fc51aa1b790d46e388b90ac9f52f/packages/kbn-unified-data-table/src/components/custom_toolbar/render_custom_toolbar.tsx#L51

stratoula commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/221905)
* Commit: 98981f0e0e8fb8547458506d2fb23cd95e40bd25

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-28c7-441c-bb19-ce94b44b32c6)
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-28c9-4ee6-9614-1f6a3fd1e575)
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd38-4751-41a3-819e-678d460edb42)
* [FTR Configs #44](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-2936-49df-9821-016867cedc81)
* [FTR Configs #100](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-29a1-4fb3-8e03-2b6a4f6a7ee6)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-28c7-441c-bb19-ce94b44b32c6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/221905/jobs/0190bd17-28c7-441c-bb19-ce94b44b32c6/artifacts/0190bd38-f257-42c7-bc54-475ab11e2f92) Jest Tests #<span></span>1 / DataTableColumnHeader should render a correct token
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-2936-49df-9821-016867cedc81) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/221905/jobs/0190bd17-2936-49df-9821-016867cedc81/artifacts/0190bd3b-549f-4071-b230-0e2f04dacc60) FTR Configs #<span></span>44 / discover discover field visualize button should save and edit chart in the dashboard on the fly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/221905#0190bd17-29a1-4fb3-8e03-2b6a4f6a7ee6) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/221905/jobs/0190bd17-29a1-4fb3-8e03-2b6a4f6a7ee6/artifacts/0190bd3d-c115-4871-8d26-f6164f2b3124) FTR Configs #<span></span>100 / lens app - group 3 lens inline editing tests should allow inline editing of a by value visualization

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/221905","number":221905,"commit":"98981f0e0e8fb8547458506d2fb23cd95e40bd25"}],"number":221905}
buildkite-pr-comment-->

stratoula commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222157)
* Commit: 92fc6dce46063307de578fb19fca88ff102c980f

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/222157#0190c188-1847-4e6c-bd45-ee28ba1165dd)

### History
* :broken_heart: [Build #221905](https://buildkite.com/elastic/kibana-pull-request/builds/221905) failed 98981f0e0e8fb8547458506d2fb23cd95e40bd25

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222157","number":222157,"commit":"92fc6dce46063307de578fb19fca88ff102c980f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/221905","number":221905,"commit":"98981f0e0e8fb8547458506d2fb23cd95e40bd25"}],"number":222157}
buildkite-pr-comment-->

stratoula commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222159)
* Commit: 89cb3e9de7b225488b9338733ff80abc0257a982

### Failed CI Steps
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/222159#0190c1b1-824c-4ff4-a9be-49cbd76b3d92)
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/222159#0190c1d2-fc24-4a79-a492-3e9aeba9518e)

### History
* :broken_heart: [Build #222157](https://buildkite.com/elastic/kibana-pull-request/builds/222157) failed 92fc6dce46063307de578fb19fca88ff102c980f
* :broken_heart: [Build #221905](https://buildkite.com/elastic/kibana-pull-request/builds/221905) failed 98981f0e0e8fb8547458506d2fb23cd95e40bd25

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222159","number":222159,"commit":"89cb3e9de7b225488b9338733ff80abc0257a982"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222157","number":222157,"commit":"92fc6dce46063307de578fb19fca88ff102c980f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/221905","number":221905,"commit":"98981f0e0e8fb8547458506d2fb23cd95e40bd25"}],"number":222159}
buildkite-pr-comment-->

stratoula commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/222230)
* Commit: 8c5a28734eea505e56f219009cf3b746a1e7f6ee

### Failed CI Steps
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/222230#0190c2c7-ecfc-4dc4-91e9-fedca4c1f208)
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/222230#0190c2e5-db80-41cf-a132-7a38869fee08)

### History
* :broken_heart: [Build #222159](https://buildkite.com/elastic/kibana-pull-request/builds/222159) failed 89cb3e9de7b225488b9338733ff80abc0257a982
* :broken_heart: [Build #222157](https://buildkite.com/elastic/kibana-pull-request/builds/222157) failed 92fc6dce46063307de578fb19fca88ff102c980f
* :broken_heart: [Build #221905](https://buildkite.com/elastic/kibana-pull-request/builds/221905) failed 98981f0e0e8fb8547458506d2fb23cd95e40bd25

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222230","number":222230,"commit":"8c5a28734eea505e56f219009cf3b746a1e7f6ee"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222159","number":222159,"commit":"89cb3e9de7b225488b9338733ff80abc0257a982"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/222157","number":222157,"commit":"92fc6dce46063307de578fb19fca88ff102c980f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/221905","number":221905,"commit":"98981f0e0e8fb8547458506d2fb23cd95e40bd25"}],"number":222230}
buildkite-pr-comment-->

stratoula commented:
@andreadelrio thanx a lot for your review

<img width="1255" alt="image" src="https://github.com/user-attachments/assets/c8f951bc-1c7b-4db5-b2a4-8e89200ca740">

As I mention above I have addressed everything except from the b. This is something which also works like that currently for the vis configuration settings and we are going to handle it most possibly in 8.17. 

One note: I added the Open In Discover link but next to the columns sort button as the Discover toolbar has already a setting to add a new control there. Are you ok with it? Otherwise I will need to create a new position and will make the component a little bit more complicated.

stratoula commented:
/ci

stratoula commented:
/ci

elasticmachine commented:
Pinging @elastic/kibana-visualizations (Team:Visualizations)

elasticmachine commented:
Pinging @elastic/kibana-esql (Team:ESQL)

dej611 reviewed (commented):
Left some code review comments.
The `lens_configuration_flyout` component was already messy and now getting even messier with all those state variables.

Playing a bit with it I've noticed some drawbacks of the resizing feature.
On resizing the chart panel keeps its proportion vs the rest of the dashboard... is it really useful this behaviour?

<img width="1500" alt="Screenshot 2024-07-23 at 17 49 56" src="https://github.com/user-attachments/assets/30915a3b-a364-4d10-834c-4d2ca98d3f98">

Here the panel has been resized to be 1/2 of the available width and the chart now is tiny at the point that it proves to be not so useful.
This can be stretched to the point where it's really just bad

<img width="1502" alt="Screenshot 2024-07-23 at 17 57 20" src="https://github.com/user-attachments/assets/a5ff5fc2-9140-421a-836a-c9e7e56e1304">

With the design changes in this PR the editing panel is getting too crowded vertically, and I feel that resizing doesn't solve it properly.
While resizing makes enough spaces for the editor to become useful, the whole experience becomes less useful when the content underneath is still part of the context: I think a proper design here should take into account both the editor and the panel when resizing/change the UI here.

andreadelrio commented:
> <img alt="image" width="1255" src="https://private-user-images.githubusercontent.com/17003240/350870617-c8f951bc-1c7b-4db5-b2a4-8e89200ca740.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3NTgxNDYsIm5iZiI6MTcyMTc1Nzg0NiwicGF0aCI6Ii8xNzAwMzI0MC8zNTA4NzA2MTctYzhmOTUxYmMtMWM3Yi00ZGI1LWIyYTQtOGU4OTIwMGNhNzQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDE4MDQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliZTRkYThmZjc3ZDg2OWMyZDRlODY3ZmIwZDc4NjFiY2FkMjRjNzU3ZjczMmUwODQzOTUyYWEwNDEzYjRkYzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2ZJz10Y6EXHdMpda5TJO7aaOeOtiUkq2cDDDxB3bwhw">

> One note: I added the Open In Discover link but next to the columns sort button as the Discover toolbar has already a setting to add a new control there. Are you ok with it? Otherwise I will need to create a new position and will make the component a little bit more complicated.

@stratoula Thanks for making those changes. This is looking good! I think the `Open in Discover` link location is not ideal but it is acceptable. After running this locally, here's another round of suggestions:

<img width="1080" alt="Frame 6" src="https://github.com/user-attachments/assets/6363e87b-9dcc-47a4-ab02-7aeaf4b5d666">

- [ ] Let's keep the blue styling for the `Open in Discover` link

I'm aware that when you set `target="_blank"` **EuiLink** automatically adds the `external` icon to the right of link but I really think this is unnecessary in this case where we're already tight for free pixels. 
- [ ] Using CSS, hide the second **EuiIcon** inside **EuiLink** (something like `svg:nth-of-type(2)`)

- [ ] Remove the `dot` at the end of `Open in Discover.`
- [ ] Add the horizontal separator below the `ES|QL Query Results` accordion

### **Regarding resizability**

- [ ] Add a `max-width` to flyout, suggest either 800px or 848px. Please use EUI size variables to define this value.
- [ ] Add a `min-width` to flyout: we probably don't want this to resize to less than the default width.

### **Other**

When checking this on my external monitor, I noticed that it looks like given the increased viewport height we are showing **white space below the results**. Can we get rid of it?

<img width="1000" alt="Frame 7" src="https://github.com/user-attachments/assets/993b50f3-d1d5-4b6c-a7ac-056d998ba4b2">

andreadelrio commented:
This is outside the scope of this PR but wanted to mention for awareness. If we preserve the current behavior (which @dej611 has concerns for), one way to increase usability is to make it easier for users to `maximize` (and then `minimize`) the panel while they have the Lens inline editor open. After [hover actions](https://github.com/elastic/kibana/pull/182535) are implemented, we can have the `maximize` action as the only action available while the user has the editor open. Also, we should make it visible at all times (not only on hover). This pattern (i.e. surfacing certain actions when the editor is open) is also what we're planning on doing for the new [inline markdown editor](https://github.com/elastic/kibana/issues/188910).

![CleanShot 2024-07-25 at 15 03 07](https://github.com/user-attachments/assets/9b531770-a594-4bb3-96da-ccb890763f78)
![image](https://github.com/user-attachments/assets/3e7321e0-0406-4a0c-bfb5-dd84acf9c09a)

https://github.com/elastic/kibana/issues/187864#issuecomment-2253260591

cc @ThomThomson

stratoula commented:
Thanx @andreadelrio, yes is not at the scope of this PR (lovely idea btw). I am going to return to this PR hopefully this week. I am dealing with other tasks with greater priority so this is mostly a PR I am working on my free time.

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/helpers.ts:
```diff
@@ -7,41 +7,82 @@
+  columns: DatatableColumn[];
+}
+
+export const getGridAtrrs = async (
```
Done https://github.com/elastic/kibana/pull/186875/commits/8ff2e6b3a54c6906c0243230056f8f25ef169aed

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.test.tsx:
```diff
@@ -196,6 +270,14 @@ describe('LensEditConfigurationFlyout', () => {
+    renderConfigFlyout({
+      canEditTextBasedQuery: true,
+    });
+    await waitFor(() => expect(screen.getByTestId('ESQLQueryResults')).toBeInTheDocument());
```
Not very helpful comment

mbondyra replied:
All good here, @mbondyra approves 👏🏼

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx:
```diff
@@ -128,6 +135,26 @@ export function LensEditConfigurationFlyout({
+          query,
+          adHocDataViews,
+          startDependencies,
+          undefined
```
Done https://github.com/elastic/kibana/pull/186875/commits/1e874dccf01444001feb1d3d9aca7ce7d46490f4

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx:
```diff
@@ -479,6 +504,77 @@ export function LensEditConfigurationFlyout({
+            `}
+                  >
+                    <h5>
+                      {i18n.translate('xpack.lens.config.ESQLQueryResults', {
```
Done https://github.com/elastic/kibana/pull/186875/commits/93534843133cf8ebc2210b77d961a8b39c69799b

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/helpers.ts:
```diff
@@ -7,41 +7,82 @@
 import { suggestionsApi } from '../../../lens_suggestions_api';
 
+export interface ESQLDataGridAttrs {
+  values: ESQLRow[];
```
Done https://github.com/elastic/kibana/pull/186875/commits/93534843133cf8ebc2210b77d961a8b39c69799b

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx:
```diff
@@ -86,7 +88,9 @@ export function LensEditConfigurationFlyout({
   const [isSuggestionsAccordionOpen, setIsSuggestionsAccordionOpen] = useState(false);
+  const [isESQLResultsAccordionOpen, setIsESQLResultsAccordionOpen] = useState(false);
   const [isVisualizationLoading, setIsVisualizationLoading] = useState(false);
+  const [dataGridAttrs, setDataGridAttrs] = useState<ESQLDataGridAttrs | undefined>(undefined);
```
I will create a tech debt issue for that, let's not do that in this PR  https://github.com/elastic/kibana/issues/189351

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/helpers.ts:
```diff
@@ -7,41 +7,82 @@
+
+  const dataView = dataViewSpec
+    ? await deps.dataViews.create(dataViewSpec)
+    : await getESQLAdHocDataview(query.esql, deps.dataViews);
```
Done https://github.com/elastic/kibana/pull/186875/commits/40e282b5316ebba25a24715fca800d0f0298e748

stratoula commented:
@andreadelrio about this:

> When checking this on my external monitor, I noticed that it looks like given the increased viewport height we are showing white space below the results. Can we get rid of it?
It is being added by the `euiDataGrid__virtualized` so it seems I can't get rid of it (unless someone else knows the eui datagrid better than me)

stratoula commented:
@andreadelrio I addressed your comments, can you give another look? 🙏 

@elastic/kibana-visualizations same, I addressed Marco's comments, I know his out but I would appreciate a followup review 🙇‍♀️

stratoula commented:
/ci

stratoula commented:
/ci

mbondyra commented on src/plugins/esql_datagrid/public/data_grid.tsx:
```diff
@@ -47,7 +59,8 @@ const DataGrid: React.FC<ESQLDataGridProps> = (props) => {
   );
-  const [rowHeight, setRowHeight] = useState<number>(5);
+  const [rowHeight, setRowHeight] = useState<number>(props.initialRowHeight ?? 5);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
```
this constant is used here and then again in line 212. Could we export it to the variable? The one above too:
```
 const DEFAULT_INITIAL_ROW_HEIGHT = 5
 const DEFAULT_ROWS_PER_PAGE = 10

stratoula replied:
Sure https://github.com/elastic/kibana/pull/186875/commits/e5f0fdc35ca5d6be0ef96e67189cabeb16d09a8c

mbondyra commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx:
```diff
@@ -128,6 +135,30 @@ export function LensEditConfigurationFlyout({
     return () => s?.unsubscribe();
   }, [dispatch, output$, layers]);
 
+  useEffect(() => {
```
how is this needed here?

stratoula replied:
It is because you need to initialize it, otherwise the accordion is not visible

stratoula commented on x-pack/plugins/lens/public/app_plugin/shared/edit_on_the_fly/lens_configuration_flyout.tsx:
```diff
@@ -479,6 +508,78 @@ export function LensEditConfigurationFlyout({
               />
             </EuiFlexItem>
           )}
+          {isOfAggregateQueryType(query) && canEditTextBasedQuery && dataGridAttrs && (
```
Hmmm, we don't do it for the other accordion items 🤔

stratoula replied:
Done here https://github.com/elastic/kibana/pull/186875/commits/b64dc30ca5d7f1570f81ab869135655ad5e75e86. As I sais you don't gain a lot but it is not bad either!

mbondyra reviewed (approved):
LGTM, tested in Chrome 👌🏼

andreadelrio reviewed (approved):
@stratoula 🚀 Amazing work here! Thanks for working on all the UI polish tasks. Just one nit, let's make the icon in the `Open in Discover is blue` as well (right now it shows in teal and black). Approving now assuming you can take care of that.

<img width="358" alt="image" src="https://github.com/user-attachments/assets/b7922de7-7fc3-4678-a2c0-504c921163fd">


---
Related task (improve preview) is already being tracked here.
https://github.com/elastic/kibana/issues/187864#issuecomment-2253260591

stratoula commented:
@andreadelrio thanx a lot for your help with the design here!

Done
<img width="534" alt="image" src="https://github.com/user-attachments/assets/c23ff952-5efb-49d5-91b1-14a8e8fef0f8">

stratoula commented:
/ci

vadimkibana reviewed (approved):
ES|QL code changes LGTM.
